app/master/slave_allocator.py

```diff
@@ -0,0 +1,57 @@
+from app.util.log import get_logger
+from app.util.ordered_set_queue import OrderedSetQueue
+from app.util.safe_thread import SafeThread
+
+
+class SlaveAllocator(object):
+    """
+    The SlaveAllocator class is responsible for allocating slaves to prepared builds.
+    """
+
+    def __init__(self, build_request_handler):
+        """
+        :type build_request_handler: BuildRequestHandler
+        """
+        self._logger = get_logger(__name__)
```

Comment:
Hmm.  I know we've had the 'slave_allocation' prefix from previous code, but now that the functions are in a SlaveAllocator class, it seems a bit overly verbose to keep it. I think _slave_allocation_worker_thread can just be changed to _worker_thread, and _slave_allocation_loop can just be something like _process_builds or something.